src/AspNetCoreModuleV2/CommonLib/BaseOutputManager.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#pragma once
+#include "IOutputManager.h"
+#include "StdWrapper.h"
+#include "EventLog.h"
+#include "exceptions.h"
+
+class BaseOutputManager :
+    public IOutputManager
+{
+public:
+    BaseOutputManager() : BaseOutputManager(/* fEnableNativeLogging */ true) {}
+    BaseOutputManager(bool enableNativeLogging) :
+        m_disposed(false),
+        stdoutWrapper(nullptr),
+        stderrWrapper(nullptr),
+        m_enableNativeRedirection(enableNativeLogging)
+    {
+        InitializeSRWLock(&m_srwLock);
+    }
+    ~BaseOutputManager() {}
+
+    void
+        StartRedirectionWithExceptionMessage(
+            std::wstring exceptionMessage)
+    {
+        auto startLambda = [](BaseOutputManager* outputManager) { outputManager->Start(); };
+
+        TryOperation(startLambda, exceptionMessage);
+    }
+
+    void
+        StopRedirectionWithExceptionMessage(std::wstring exceptionMessage)
+    {
+        auto stopLambda = [](BaseOutputManager* outputManager) { outputManager->Stop(); };
+
+        TryOperation(stopLambda, exceptionMessage);
+    }
+
+protected:
+    std::wstring m_stdOutContent;
+    bool m_disposed;
+    bool m_enableNativeRedirection;
+    SRWLOCK m_srwLock{};
+    std::unique_ptr<StdWrapper> stdoutWrapper;
+    std::unique_ptr<StdWrapper> stderrWrapper;
+
+    void
+        TryOperation(void(*func)(BaseOutputManager* outputManager),
+            std::wstring exceptionMessage)
+    {
+        try
+        {
+            func(this);
```

Comment:
You can capture `this` in lambda.

Reply:
not sure how to work around: https://stackoverflow.com/questions/31611415/converting-a-lambda-expression-with-variable-capture-to-a-function-pointer

Reply:
Did you try using templated function?

```
template<typename Functor>
void f(Func ff)
{
ff()?
}
```